klasses/views.py

```diff
@@ -0,0 +1,76 @@
+"""
+Views for Klasses app
+"""
+import logging
+
+from django.contrib.auth.models import User
+from rest_framework import (
+    authentication,
+    permissions,
+    status,
+)
+from rest_framework.generics import get_object_or_404
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from backends.edxorg import EdxOrgOAuth2
+from klasses.api import get_user_klasses_info, get_user_klass_info
+from klasses.permissions import CanReadIfSelf
+
+log = logging.getLogger(__name__)
+
+
+class UserKlassList(APIView):
+    """
+    Class based view for user klasse list view.
+    """
+    authentication_classes = (
+        authentication.SessionAuthentication,
+    )
+    permission_classes = (
+        permissions.IsAuthenticated,
+        CanReadIfSelf,
+    )
+
+    def get(self, request, username, *args, **kargs):  # pylint: disable=unused-argument
+        """
+        Returns information about the payments for the user in all the klasses she is allowed to pay.
+        """
+        user = get_object_or_404(
+            User,
+            social_auth__uid=username,
+            social_auth__provider=EdxOrgOAuth2.name
+        )
+
+        return Response(get_user_klasses_info(user=user))
+
+
+class UserKlassDetail(APIView):
```

Comment:
Actually, I do not deal with the klass object at all in this view, so I prefer to leave as is